hera_sim/rfi.py

```diff
@@ -184,9 +189,65 @@ def rfi_scatter(fqs, lsts, rfi=None, chance=0.0001, strength=10, std=10):
     """
     if rfi is None:
         rfi = np.zeros((lsts.size, fqs.size), dtype=np.complex)
-    assert rfi.shape == (lsts.size, fqs.size)
+    assert rfi.shape == (lsts.size, fqs.size), "rfi shape is not (lsts.size, fqs.size)"
+
     rfis = np.where(np.random.uniform(size=rfi.size) <= chance)[0]
     rfi.flat[rfis] += np.random.normal(strength, std) * np.exp(
-        1j * np.random.uniform(size=rfis.size)
+        2 * np.pi * 1j * np.random.uniform(size=rfis.size)
     )
     return rfi
+
+
+def rfi_dtv(fqs, lsts, rfi=None, freq_min=.174, freq_max=.214, width=0.008,
+            chance=0.0001, strength=10, strength_std=10):
+    """
+    Generate an (NTIMES, NFREQS) waterfall containing Digital TV RFI.
+
+    DTV RFI is expected to be of uniform bandwidth (eg. 8MHz), in contiguous
+    bands, in a nominal frequency range. Furthermore, it is expected to be
+    short-duration, and so is implemented as randomly affecting discrete LSTS.
+
+    There may be evidence that close times are correlated in having DTV RFI,
+    and this is *not currently implemented*.
+
+    Args:
+        fqs (array-like): shape=(NFREQS,), GHz
+            the spectral frequencies of the waterfall to be generated.
+        lsts (array-like): shape=(NTIMES,), radians
+            local sidereal times of the waterfall to be generated.
+        rfi (array-like): shape=(NTIMES,NFREQS), default=None
+            an array to which the RFI will be added.  If None, a new array
+            is generated.
+        freq_min, freq_max (float):
+            the min and max frequencies of the full DTV band [GHz]
+        width (float):
+            Width of individual DTV bands [GHz]
+        chance (float): default=0.0001
+            the probability that a time/freq bin will be assigned an RFI impulse
+        strength (float): Jy, default=10
+            the average amplitude of the spike generated in each time/freq bin
+        strength_std (float): Jy, default = 10
+            the standard deviation of the amplitudes drawn for each time/freq bin
+    Returns:
+        rfi (array-like): shape=(NTIMES,NFREQS)
+            a waterfall containing RFI
+    """
+    if rfi is None:
+        rfi = np.zeros((lsts.size, fqs.size), dtype=np.complex)
+    assert rfi.shape == (lsts.size, fqs.size), "rfi shape is not (lst.size, fqs.size)"
+
+    bands = np.arange(freq_min, freq_max, width)  # lower freq of each potential DTV band
+
+    delta_f = fqs[1] - fqs[0]
+
+    for band in bands:
+        fq_ind_min = np.argwhere(band >= fqs)[0][0]
+        fq_ind_max = fq_ind_min + int(width / delta_f)
+        this_rfi = rfi[:, fq_ind_min:fq_ind_max]
+
+        rfis = np.where(np.random.uniform(size=this_rfi) <= chance)[0]
```

Comment:
I think this results in each channel within a band acting independently. But I think we should have a given band either all on or all off at the same time.

Reply:
This is a very good catch @adampbeardsley. I _think_ if you set up a boolean index on the lsts like so:

```suggestion
        rfis = np.random.uniform(size=lsts.size) <= chance
```

and then replace line 249 with

```
# shape of subarray that is called by boolean index
sz = [np.count_nonzero(rfis), int(width / delta_f)]
this_rfi[rfis] += np.random.normal(strength, strength_std, size=sz) * np.exp(
	            2 * np.pi * 1j * np.random.uniform(size=sz)
```

that ought to work. Please let me know your thoughts.

Reply:
I'm worried about the second component of the sz vector. Might need to add 1 to it.

Reply:
I've addressed this in 342b0a697db2f8afa857058af6da6bf3dbbd8c7a